packages/create-toolpad-app/src/generateProject.ts

```diff
@@ -42,63 +42,67 @@ export default function generateProject(
   }
     `;
 
-  const dashboardLayoutContent = `
+  const dashboardLayoutContent = `import * as React from 'react';
   import { DashboardLayout } from '@toolpad/core/DashboardLayout';
-
-  export default function Layout({
-    children,
-  }: Readonly<{ children: React.ReactNode }>) {
-    return (
-      <DashboardLayout>{children}</DashboardLayout>
-    );
-  }
-  `;
-
-  const dashboardPageLayout = `
   import { PageContainer } from '@toolpad/core/PageContainer';
-
-  export default function Layout({
-    children,
-  }: Readonly<{ children: React.ReactNode }>) {
+  
+  export default function Layout(props: { children: React.ReactNode }) {
     return (
-      <PageContainer>{children}</PageContainer>
+      <DashboardLayout>
+        <PageContainer>{props.children}</PageContainer>
+      </DashboardLayout>
     );
-  }
+  }  
   `;
 
-  const rootPageContainer = `
-  import Link from "next/link";
-  import { Button, Container, Typography, Box } from "@mui/material";
-
+  const rootPageContent = `import Link from 'next/link';
+  import { Container, Typography, Box } from '@mui/material';
+  import NavigateButton from './NavigateButton';
+  
   export default function Home() {
     return (
       <Container>
         <Box sx={{ my: 4 }}>
           <Typography variant="h4" component="h1" gutterBottom>
-            Welcome to{" "}
-            <Link href="https://mui.com/toolpad/core/introduction">
-              Toolpad Core!
-            </Link>
+            Welcome to <Link href="https://mui.com/toolpad/core/introduction">Toolpad Core!</Link>
           </Typography>
-
+  
           <Typography variant="body1">
             Get started by editing <code>(dashboard)/page/page.tsx</code>
           </Typography>
-
-          <Box sx={{ mt: 2 }}>
-            <Link href="/page">
-              <Button variant="contained" color="primary">
-                Go to Page
-              </Button>
-            </Link>
-          </Box>
+          <NavigateButton />
         </Box>
       </Container>
     );
-  }
+  }  
   `;
 
-  const dashboardPage = `
+  const navigateButtonContent = `'use client';
+import * as React from 'react';
+import Link from 'next/link';
+import { LoadingButton } from '@mui/lab';
+import { Box } from '@mui/material';
+
+export default function NavigateButton() {
```

Comment:
I think it doesn't make sense to make this a separate file? Seems to specific, better for it to be in the intro page only?

Reply:
This is a client component, so – I could be wrong  – it must be in its own file with "use client" at the top

Reply:
I think you can have `'use client'` as the first line inside the component?

But if in terms of file structure you think this makes more sense anyway it's not a big deal!